From 58258916cdf15e8e9ce6094f013aad89d86c0158 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 18 Jun 2014 20:04:19 -0400
Subject: [block] blk-mq: merge blk_mq_drain_queue and __blk_mq_drain_queue

Message-id: <1403121864-7956-3-git-send-email-snitzer@redhat.com>
Patchwork-id: 84265
O-Subject: [RHEL7.1 PATCH 153/150] blk-mq: merge blk_mq_drain_queue and __blk_mq_drain_queue
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 95ed068165d8edf6a81f9a73df0bb05c38602c3c
Author: Christoph Hellwig <hch@lst.de>
Date:   Fri Jun 13 19:43:35 2014 +0200

    blk-mq: merge blk_mq_drain_queue and __blk_mq_drain_queue

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index fd8b485..0ef2dc7 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -109,7 +109,7 @@ static void blk_mq_queue_exit(struct request_queue *q)
  __percpu_counter_add(&q->mq_usage_counter, -1, 1000000);
 }
 
-static void __blk_mq_drain_queue(struct request_queue *q)
+void blk_mq_drain_queue(struct request_queue *q)
 {
  while (true) {
   s64 count;
@@ -139,12 +139,7 @@ static void blk_mq_freeze_queue(struct request_queue *q)
  spin_unlock_irq(q->queue_lock);
 
  if (drain)
-  __blk_mq_drain_queue(q);
-}
-
-void blk_mq_drain_queue(struct request_queue *q)
-{
- __blk_mq_drain_queue(q);
+  blk_mq_drain_queue(q);
 }
 
 static void blk_mq_unfreeze_queue(struct request_queue *q)
-- 
1.7.1